esp-hal-common/src/gpio.rs

```diff
@@ -51,6 +51,32 @@ pub struct Input<MODE> {
     _mode: PhantomData<MODE>,
 }
 
+pub struct InvertedInput<MODE> {
+    _mode: PhantomData<MODE>,
+}
+
+trait InputMode {
+    fn inverted() -> bool;
```

Comment:
You may want to explain what these traits are, it took me a few seconds to figure it out.

Reply:
Good point about some explanation. Also agree on associated constants - expresses the intent better (while I would expect the compiler to produce the same code depending on opt-level) 👍

Reply:
Thanks! :) Inverted here means that if I set a pin to high, the actual output is low, and conversely, I read low inputs as high, right?

Reply:
Yes - but only when pins are connected to a peripheral. For usual GPIO operations this is not supported but then software could invert the state itself. It's an edge case t.b.h.

Reply:
Sweet! IIRC other GPIO modes have `From<GpioPin<Unknown, GPIONUM>>` implemented for them. Do you think that would make sense here, too?

Reply:
Probably yes - at least just completeness